File: split/printskipparam.pas
procedure printskipparam(n: integer);
begin
  case n of
    0: printesc(376);
    1: printesc(377);
    2: printesc(378);
    3: printesc(379);
    4: printesc(380);
    5: printesc(381);
    6: printesc(382);
    7: printesc(383);
    8: printesc(384);
    9: printesc(385);
    10: printesc(386);
    11: printesc(387);
    12: printesc(388);
    13: printesc(389);
    14: printesc(390);
    15: printesc(391);
    16: printesc(392);
    17: printesc(393);
    others: print(394)
  end;
end;

File: cpp/printskipparam.cpp
#include "printskipparam.h"
#include "printesc.h"
#include "print.h"

void printskipparam(int n)
{
	switch (n)
	{
		case 0: 
			printesc("lineskip");
			break;
		case 1: 
			printesc("baselineskip");
			break;
		case 2: 
			printesc("parskip");
			break;
		case 3: 
			printesc("abovedisplayskip");
			break;
		case 4: 
			printesc("belowdisplayskip");
			break;
		case 5: 
			printesc("abovedisplayshortskip");
			break;
		case 6: 
			printesc("belowdisplayshortskip");
			break;
		case 7: 
			printesc("leftskip");
			break;
		case 8: 
			printesc("rightskip");
			break;
		case 9: 
			printesc("topskip");
			break;
		case 10: 
			printesc("splittopskip");
			break;
		case 11: 
			printesc("tabskip");
			break;
		case 12: 
			printesc("spaceskip");
			break;
		case 13: 
			printesc("xspaceskip");
			break;
		case 14: 
			printesc("parfillskip");
			break;
		case 15: 
			printesc("thinmuskip");
			break;
		case 16: 
			printesc("medmuskip");
			break;
		case 17: 
			printesc("thickmuskip");
			break;
		default: 
			print("[unknown glue parameter!]");
	}
}
